### PR TITLE
14035 batchinsert support ignore

### PR DESCRIPTION
**Pull request by qihuajun**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14035

### CODE REVIEW

File: framework/db/cubrid/QueryBuilder.php [hunk 1]
```diff
@@ -193,4 +193,12 @@ private function getColumnDefinition($table, $column)
         }
         return null;
     }
+
+    /**
+     * @inheritdoc
+     */
+    public function batchInsert($table, $columns, $rows, $ignore = false, $replace = false)
+    {
+        return parent::batchInsert($table, $columns, $rows, false, false);
```
Comment (MysteryDragon) [inline framework/db/cubrid/QueryBuilder.php]: Maybe it's better to throw exception in such cases?

Comment (rossoneri) [inline framework/db/cubrid/QueryBuilder.php]: add exceptions

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,13 @@ Yii Framework 2 Change Log
 
 2.0.12 under development
 --------------------------
+- Enh #14035: Support `INSERT IGNOTE INTO` or `REPLACE INTO` for MySQL while execute batchInsert
+- Bug #14012: `yii\db\pgsql\Schema::findViewNames()` was skipping materialized views (insolita)
```
Comment (samdark) [inline framework/CHANGELOG.md]: Seems unrelated.

File: framework/db/Command.php [hunk 3]
```diff
@@ -455,11 +455,13 @@ public function insert($table, $columns)
      * @param string $table the table that new rows will be inserted into.
      * @param array $columns the column names
      * @param array $rows the rows to be batch inserted into the table
+     * @param boolean $ignore whether to excute insert ignore into, only support MySQL
+     * @param boolean $replace whether to excute `repace into` instead of `insert into` , only support MySQL
      * @return $this the command object itself
      */
-    public function batchInsert($table, $columns, $rows)
+    public function batchInsert($table, $columns, $rows, $ignore = false, $replace = false)
```
Comment (samdark) [inline framework/db/Command.php]: Method signature change breaks backwards compatibility so with it the pull request could be merged into 2.1 only.

File: framework/CHANGELOG.md (hunk 2 above)
Comment (rossoneri) [inline framework/CHANGELOG.md]: Sorry,the second line was merged from upstream.

File: framework/db/Command.php (hunk 3 above)
Comment (rossoneri) [inline framework/db/Command.php]: ok!

**Comment by SilverFire** [comment]
`ignore` and `replace` options can not be used together.
I think it's better to create separate methods:

 - `batchInsertIgnore($table, $columns, $rows)`
 - `batchInsertReplace($table, $columns, $rows)`

Then we can deliver this feature without BC breaking and better semantic.

**Comment by Faryshta** [comment]
@SilverFire with that approach it would break bc and can be added to 2.0 branch?

it can also save a lot of code by building a `UnsupportedInsertIgnore` trait to avoid repeating the same exception over and over

**Comment by rossoneri** [comment]
@SilverFire Update My code according to your advice. Actually I thought that way at the beginning.

**Comment by samdark** [comment]
Won't make it into 2.0.
